rir/src/compiler/translations/pir_2_rir/pir_2_rir.cpp

```diff
@@ -706,12 +706,12 @@ size_t Pir2Rir::compileCode(Context& ctx, Code* code) {
             order.push_back(bb->id);
     });
 
-    std::unordered_map<Instruction*, bool> needsEnsureNamed;
+    std::unordered_map<Instruction*, size_t> needsRefcount;
     {
         StaticReferenceCount analysis(cls, log);
         for (auto& u : analysis.result().uses) {
             if (u.second > 1)
-                needsEnsureNamed[u.first] = u.second;
+                needsRefcount[u.first] = u.second - 1;
```

Comment:
If an instruction is used multiple times, but not by subassign, does it still need `setShared`?